librz/util/buf.c

```diff
@@ -1406,6 +1463,18 @@ RZ_API ut64 rz_buf_fwd_scan(RZ_NONNULL RzBuffer *b, ut64 start, ut64 amount, RZ_
 	return addr - start;
 }
 
+/**
+ * \brief To get whole buffer for scanning in hot paths.
+ *
+ * \param buffer RzBuffer to read
+ * \param sz Size of returned data
+ * \return ut8* pointer to whole buffer
+ */
+RZ_API ut8 *rz_buf_get_whole_hot_paths(RzBuffer *b, RZ_NONNULL RZ_OUT ut64 *sz) {
```

Comment:
```suggestion
/**
 * \brief Get the whole buffer for scanning in hot paths.
 * Please use `rz_buf_fwd_scan()` or `rz_buf_read()`, if your use case is NOT performance critical.
 *
 * \param buffer RzBuffer to read.
 * \param sz Size of returned data in bytes.
 * \return Immutable pointer to the whole buffer or NULL in case of failure.
 */
RZ_API const ut8 *rz_buf_get_whole_hot_paths(RZ_BORROW RzBuffer *b, RZ_NONNULL RZ_OUT ut64 *sz) {
```